zen_document_parser/base.py

```diff
@@ -0,0 +1,231 @@
+from enum import IntEnum
+from pdfquery import PDFQuery
+from pdfminer.pdfdocument import PDFSyntaxError
+
+from zen_document_parser.exceptions import InvalidPDFError, UnknownVariantError
+
+
+class DocFieldType(IntEnum):
+    TEXT = 1
+    NUMBER = 2
+    CUSTOM = 3  # TODO: Forget this and have 'type' take a callable instead?
+
+
+class DocField(object):
+    def __init__(self, bbox, type=DocFieldType.TEXT, required=False,
+                 description=None):
+        self.bbox = bbox
+        self.type = type
+        self.required = required
+        self.description = description
+
+
+class DocVariantMeta(type):
```

Comment:
@kgritesh @suraj-arya 

Ok, I was thinking about the dict approach, for which the above is an argument for.

You were talking about just setting them as attributes on the instance (ie. processing the DocFields in __init__).

I thought I had a few good points that favored the use of metaclasses. But now that I think of it, all those things can be accomplished in the other approach as well.

Guess I got carried away in the flow of learning how metaclasses work :)